doc/source/whatsnew/v0.17.0.txt

```diff
@@ -26,7 +26,8 @@ New features
 
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
-
+- `regex` argument to DataFrame.filter now handles numeric column names instead of raising an exception.
```

Comment:
add the issue number (this PR number) onto the end (see how the other issues are done)